pkg/controller/jaeger/elasticsearch.go

```diff
@@ -83,6 +83,21 @@ func waitForAvailableElastic(c client.Client, es esv1.Elasticsearch) error {
 				available++
 			}
 		}
+		ssList := corev1.StatefulSetList{}
+		if err = c.List(context.Background(), client.MatchingLabels(labels).InNamespace(es.Namespace), &ssList); err != nil {
+			if k8serrors.IsNotFound(err) {
+				// the object might have not been created yet
+				log.WithFields(log.Fields{
+					"namespace": es.Namespace,
+					"name":      es.Name,
+				}).Debug("Elasticsearch cluster doesn't exist yet.")
+				return false, nil
+			}
+			return false, err
+		}
+		for _, d := range ssList.Items {
+			available += d.Status.ReadyReplicas
```

Comment:
That may seem a bit confusing when debugging an issue: when there's a mismatch, which ones aren't ready?

Wouldn't it be better to have two counters, one for the deployments and one for the stateful sets?

Reply:
+1 I have split them into SS and deployments.